File: Java/2181_Merge_Nodes_in_Between_Zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        if(head == null || head.next == null){
            return null;
        }
        
        ListNode result = new ListNode(0);
        ListNode cur = result;
        head = head.next;
        
        int temp = 0;
        while(head!=null){
            temp += head.val;
            if(head.val == 0){
                cur.next = new ListNode(temp);
                cur = cur.next;
                temp = 0;
            }
            head = head.next;
        }
        return result.next;
    }
}

File: Python3/2181_Merge_Nodes_in_Between_Zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        result = temp = ListNode()
        total = 0
        while head != None:
        # while (head is not None):
            if head.val == 0:
                temp.next = ListNode(total)
                temp = temp.next
                total = 0
            else:
                total += head.val
            head = head.next
        return result.next.next
